mcux/mcux-sdk/CMSIS/Core_AArch64/Include/core_common.h

```diff
@@ -69,9 +69,7 @@
 #define GET_EL(_mode)		(((_mode) >> MODE_EL_SHIFT) & MODE_EL_MASK)
 
 /* MPIDR */
-#ifndef MPIDR_AFFLVL_MASK
-#define MPIDR_AFFLVL_MASK	(0xfful)
-#endif
+#define MPIDR_AFFLVL_MASK	(0xffULL)
```

Comment:
Ok so you are matching the internal define so that the compiler does not complain about the redefinition right? I guess it's ok, certainly avoids the the include order problem but would it be an option to include the upstream header instead and not define these in the first place if that's an option?

cc @gmarull for a second opinion

Reply:
if the mask needs to be ULL then this patch is an actual bug fix, and if it luckily matches Zephyr define we can live with that. We are living with vendor HALs anyway :hankey:

Reply:
Alrighty, then I guess that's what I should have done yesterday instead of ifdef'ing it. Okay, live and learn.

Reply:
Yes, indeed, it should be ULL, and we have no building warning with this update.

Reply:
@JiafeiPan. Is there a way to avoid this duplication. Maybe move this code to the Zephyr CMSIS module?

Reply:
@mmahadevan108, in the long term, we need to upstream CMSIS and then migrate to Zephyr's CMSIS, currently we need still to use hal's CMSIS before that,  and not able to only move part of code to another CMSIS.